boards/arm/sama5/giant-board/include/board_492mhz.h

```diff
@@ -0,0 +1,207 @@
+#define BOARD_TSD_DEBOUNCE         (10000000)  /* 10 milliseconds (units nanoseconds) */
+
+/* Resulting frequencies */
+
+/* Giant Board
+ *
+ * Crystal frequency:       24 MHz
+ * CPU clock:              492 MHz
+ * Master clock:           164 MHz
+ */
+
+#define BOARD_MAINCK_FREQUENCY     BOARD_MAINOSC_FREQUENCY
+#define BOARD_PLLA_FREQUENCY       (984000000) /* PLLACK: 41 * 24Mhz / 1 */
+#define BOARD_PCK_FREQUENCY        (492000000) /* CPU:  PLLACK / 2 / 1  */
+#define BOARD_MCK_FREQUENCY        (164000000) /* MCK:  PLLACK / 1 / 1 / 3 */
+#define BOARD_ADCCLK_FREQUENCY     (10250000)  /* ADCCLK: MCK / ((7+1)*2) */
+
+/* Clocking to certain peripherals may be MCK/2.
+ *
+ * REVISIT:  I am not sure why this is.  Perhaps because of H32MXDIV?
+ */
+
+#define BOARD_PIT_FREQUENCY        (BOARD_MCK_FREQUENCY >> 1)
+#define BOARD_USART_FREQUENCY      (BOARD_MCK_FREQUENCY >> 1)
+#define BOARD_FLEXCOM_FREQUENCY    (BOARD_MCK_FREQUENCY >> 1)
+
+#if defined(CONFIG_SAMA5_EHCI) || defined(CONFIG_SAMA5_OHCI) || \
+    defined(CONFIG_SAMA5_UDPHS)
+
+/* The USB Host High Speed requires a 480 MHz clock (UPLLCK) for the embedded
+ * High-speed transceivers. UPLLCK is the output of the 480 MHz UTMI PLL
+ * (UPLL).  The source clock of the UTMI PLL is the Main OSC output:  Either
+ * the 12MHz internal RC oscillator on a an external 12MHz crystal.  The
+ * Main OSC must be 12MHz because the UPLL has a built-in 40x multiplier.
+ *
+ * For High-speed operations, the user has to perform the following:
+ *
+ *   1) Enable UHP peripheral clock, bit (1 << AT91C_ID_UHPHS) in
+ *      PMC_PCER register.
+ *   2) Write CKGR_PLLCOUNT field in PMC_UCKR register.
+ *   3) Enable UPLL, bit AT91C_CKGR_UPLLEN in PMC_UCKR register.
+ *   4) Wait until UTMI_PLL is locked. LOCKU bit in PMC_SR register
+ *   5) Enable BIAS, bit AT91C_CKGR_BIASEN in PMC_UCKR register.
+ *   6) Select UPLLCK as Input clock of OHCI part, USBS bit in PMC_USB
+ *      register.
+ *   7) Program the OHCI clocks (UHP48M and UHP12M) with USBDIV field in
+ *      PMC_USB register. USBDIV must be 9 (division by 10) if UPLLCK is
+ *      selected.
+ *   8) Enable OHCI clocks, UHP bit in PMC_SCER register.
+ *
+ * Steps 2 through 7 performed here.  1 and 8 are performed in the EHCI
+ * driver is initialized.
+ */
+
+#  define BOARD_USE_UPLL             1     /* Use UPLL for clock source */
+#  define BOARD_CKGR_UCKR_UPLLCOUNT  (15)  /* Maximum value */
+#  define BOARD_CKGR_UCKR_BIASCOUNT  (15)  /* Maximum value */
+#  define BOARD_UPLL_OHCI_DIV        (10)  /* Divide by 10 */
+#endif
+
+/* HSMCI clocking
+ *
+ * Multimedia Card Interface clock (MCCK or MCI_CK) is Master Clock (MCK)
+ * divided by (2*(CLKDIV) + CLOCKODD + 2).
+ *
+ *   MCI_SPEED = MCK / (2*CLKDIV + CLOCKODD + 2)
+ *
+ * Where CLKDIV has a range of 0-255.
+ */
+
+/* MCK = 132MHz, CLKDIV = 164,
+ * MCI_SPEED = 132MHz / (2*164 + 0 + 2) = 400 KHz
+ */
+
+#define SDMMC_INIT_CLKDIV          (164 << SDMMC_MR_CLKDIV_SHIFT)
+
```

Comment:
HSMCI is not available on the Giant board, is it?  I would need to check.  The SD/MMC interface is via SPI, isn't it?

Reply:
In the SAMA5D2, HSMCI is referred as SDMMC, isn't it?